File: Weapon.java
public class Weapon {
    public String name = "DEFAULT";
    public int damage = 0;

    public Weapon(String n, int d){
        this.name = n;
        this.damage = d;
    }
    
    public void useOn(Entity u, Entity r){ //should handle printing
        int loss = this.damage - r.armor.effectiveness;
        if(loss < 0){
            System.out.println(u.name + " used "+this.name+" on "+r.name+" but "+r.name+"'s armor was too thick!");
        }else{
            r.health -= loss;
            System.out.println(u.name + " used "+this.name+" on "+r.name+" and dealt "+loss+" damage!");
        }
    }

    public static Weapon getNumber(int id){
        switch (id){ //need cases 0 - 5 inclusive
            case 0: //starting weapon
                return new Weapon("Toothpick", 20);
            case 1:
                return new Weapon("Big Stick", 35);
            case 2:
                return new Weapon("Tiger Bone", 60);
            case 3:
                return new Weapon("Elemental Blade", 85);
            case 4: 
                return new Weapon("Flaming Halberd", 105);
            case 5:
                return new Weapon("Holy Sword", 140);
            default:
                //default weapon
                return new Weapon("DEFAULT WEAPON", 0);
        }
    }
}

File: scripts/Weapon.js
export class Weapon {

    constructor(n, d, u){
        if(n == null){
            this.name = "DEFAULT";
        } else{
            this.name = n;
        }
        if (d==null){
            this.damage = 0;
        } else {
            this.damage = d;
        }
        if(u == null){
            this.u = "./images/toothpick.jpg";
        }else{
            this.url = u;
        }
    }
    
    useOn(h, m){ //should return values
        
        let loss = Math.max(this.damage - m.armor.effectiveness, 0);        m.health -= loss;
        
        if(loss <= 0){
            return "\nThe "+m.name+" was not damaged by the "+this.name+"!";
        }else{
            return "\n"+m.name+"'s defenses were brutally penetrated by "+h.name+"'s "+this.name+" for "+loss+" damage!";
        }
    }

    static getNumber(id){
        switch (id){ 
            case 0: //starting weapon
                return new Weapon("Toothpick", 20000, "./images/toothpick.jpg");
            case 1:
                return new Weapon("Big Stick", 35, "./images/stick.png");
            case 2:
                return new Weapon("Tiger Bone", 60, "./images/tigerbone.jpg");
            case 3:
                return new Weapon("Elemental Blade", 85, "./images/elementalsword,jpg");
            case 4: 
                return new Weapon("Flaming Halberd", 105, "./images/flaminghalberd.jpg");
            case 5:
                return new Weapon("Holy Sword", 140, "./images/holysword.jpeg");
            default:
                //default weapon
                return new Weapon("DEFAULT WEAPON", 0, "");
        }
    }
}
